x/crosschain/keeper/keeper_cross_chain_tx.go

```diff
@@ -210,7 +210,39 @@ func (k Keeper) CctxAllPending(c context.Context, req *types.QueryAllCctxPending
 		sends = append(sends, &cctx)
 	}
 
-	return &types.QueryAllCctxPendingResponse{CrossChainTx: sends}, nil
+	// return scanned nonce range
+	scannedNonces := &types.PendingNonces{}
+	*scannedNonces = p
+	scannedNonces.NonceLow = startNonce
+	return &types.QueryAllCctxPendingResponse{CrossChainTx: sends, PendingNonces: scannedNonces}, nil
+}
+
+func (k Keeper) CctxAllPendingInNonceRange(c context.Context, req *types.QueryAllCctxPendingInNonceRangeRequest) (*types.QueryAllCctxPendingInNonceRangeResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+	tss, found := k.GetTSS(ctx)
+	if !found {
+		return nil, status.Error(codes.Internal, "tss not found")
+	}
+	sends := make([]*types.CrossChainTx, 0)
+
+	for i := req.NonceLow; i < req.NonceHigh; i++ {
```

Comment:
Don't we want `req.NonceHigh` to be included?
If the case, should be `i <= req.NonceHigh`

Reply:
Actually the range means [NonceLow, NonceHigh) excluding the `NonceHigh` by design.